src/daemon/rpmostreed-transaction-types.c

```diff
@@ -326,11 +326,22 @@ rollback_transaction_execute (RpmostreedTransaction *transaction,
 {
   RollbackTransaction *self = (RollbackTransaction *) transaction;
   OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
+  OstreeDeployment *booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
 
+  g_autoptr(OstreeDeployment) pending_deployment = NULL;
   g_autoptr(OstreeDeployment) rollback_deployment = NULL;
-  ostree_sysroot_query_deployments_for (sysroot, self->osname, NULL, &rollback_deployment);
-  if (!rollback_deployment)
+  ostree_sysroot_query_deployments_for (sysroot, self->osname,
+                                        &pending_deployment, &rollback_deployment);
+
+  if (!rollback_deployment && !pending_deployment) /* i.e. do we just have 1 deployment? */
     return glnx_throw (error, "No rollback deployment found");
+  else if (!rollback_deployment)
+    {
+      /* If there isn't a rollback deployment, but there *is* a pending deployment, then we
+       * want "rpm-ostree rollback" to put the currently booted deployment back on top. This
+       * also allows users to effectively undo a rollback operation. */
+      rollback_deployment = g_object_ref (booted_deployment);
```

Comment:
If we use `pending_deployment` here, we're not actually changing anything, right? The first rollback will put the previously booted deployment ahead of the currently booted one. So the second rollback should put the *currently* booted deployment ahead, right?

The check for a `pending_deployment` above just guarantees us that the booted deployment is not the current default.